resource reuse/plugins/org.polarsys.kitalpha.resourcereuse/src/org/polarsys/kitalpha/resourcereuse/registry/TargetPlatformReader.java

```diff
@@ -11,40 +11,140 @@
+ * An implementation of the {@link ModelBasedReader} focusing on retrieving information focused on Platform bundles
+ * only.
+ * 
  * @author Thomas Guiu
+ * @author <a href="mailto:arnaud.dieumegard@obeo.fr">Arnaud Dieumegard</a>
  * 
  */
 public class TargetPlatformReader extends ModelBasedReader {
+    
+    private List<String> clearedBundles = new ArrayList<String>();
+
+    @Override
+    /*
+     * In this implementation we override readResources and thus do not need the content of the getModels method. The
+     * ModelBasedReader signature may need to be updated to allow to remove this method.
+     */
+    protected IPluginModelBase[] getModels() {
+        return new IPluginModelBase[0];
+    }
+
+    @Override
+    public ResourceRegistry readResources(ResourceRegistry parent) {
+        registry = new ModelBasedRegistry(parent, this);
+        // Ensure clearedBundles list is cleaned
+        clearedBundles.clear();
+        IExtensionRegistry extensionRegistry = RegistryFactory.getRegistry();
+
+        for (IConfigurationElement confElement : extensionRegistry.getConfigurationElementsFor(CONCERN_EXTENSION_ID)) {
+            String bundleName = getBundleName(confElement);
+            String concernName = confElement.getAttribute("name");
+            String user = confElement.getAttribute("userResourceId");
+            String used = confElement.getAttribute("usedResourceId");
+
+            // Only clear once a bundle content in the registry
+            if (!clearedBundles.contains(bundleName)) {
+                registry.clear(bundleName);
+                clearedBundles.add(bundleName);
+            }
+            registry.addConcern(bundleName, concernName, user, used);
+        }
+        for (IConfigurationElement confElement : extensionRegistry.getConfigurationElementsFor(RESOURCE_EXTENSION_ID)) {
+            String bundleName = getBundleName(confElement);
+            Resource res = createResource(confElement);
+
+            // Only clear once a bundle content in the registry
+            if (!clearedBundles.contains(bundleName)) {
+                registry.clear(bundleName);
+                clearedBundles.add(bundleName);
+            }
+            registry.addResource(bundleName, res);
+        }
+
+        // Clean cleared bundles
+        clearedBundles.clear();
+        return registry;
+    }
+
+    private String getBundleName(IConfigurationElement confElement) {
+        IExtension declaringExtension = confElement.getDeclaringExtension();
+        IContributor contributor = declaringExtension.getContributor();
+        
+        String bundleName;
+        if (contributor instanceof RegistryContributor) {
+            bundleName = ((RegistryContributor)contributor).getActualName();
+        } else {
+            bundleName = contributor.getName();
+        }
+        return bundleName;
+    }
+
+    private Resource createResource(IConfigurationElement confElement) {
+        Resource res = new Resource();
```

Comment:
It better to use the resource reuse factory !!

Reply:
There is no factory, this is not some EMF-generated code

Reply:
Thanks for the verification.